core/templates/dev/head/base.html

```diff
@@ -56,7 +56,7 @@
 
     <link rel="apple-touch-icon" href="/images/logo/favicon.png">
     <title itemprop="name">{% block subtitle %}{% endblock %} - {% block maintitle %}Oppia{% endblock %}</title>
-
+    <base href="/gallery">
```

Comment:
Hmm, its because of relative links setting in `$locationProvider`. `html5Mode` shouldn't be messing with things.